gradle-jdks-excavator-configurations/src/main/java/com/palantir/gradle/jdks/GradleJdksConfigurator.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.jdks;
+
+import com.palantir.gradle.jdks.setup.common.Arch;
+import com.palantir.gradle.jdks.setup.common.Os;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+
+public final class GradleJdksConfigurator {
+
+    private static final JdkDistributions jdkDistributions = new JdkDistributions();
+
+    /**
+     * Writes the jdk configuration files for a given jdk distribution.
+     *
+     * @param targetDir the target directory where the configuration files should be written to.
+     * @param baseUrl the base url where the jdk distributions are hosted e.g. `https://corretto.aws`
+     * @param jdkDistributionName the name of the jdk distribution e.g. corretto.
+     * @param os the operating system of the jdk distribution e.g. linux.
+     * @param arch the architecture of the jdk distribution e.g. x64.
+     * @param javaVersion the major java version e.g. 8.
+     * @param jdkVersion the full version of the jdkVersion e.g. 8.282.08.1.
+     * @param caCerts the list of certificates that need to be added to the jdk. This influences the hash and the
+     * local installation directory name.
+     */
+    public static void writeGradleJdkInstallationConfigs(
```

Comment:
What do we expect to call this? If excavator how will it get all the correct data?

Reply:
It might be best if this takes the JSON representation that's somewhere else in this repo.

Reply:
[This one](https://github.com/palantir/gradle-jdks/blob/b59b494f8e29a302d70e518e2f3df4f3e9fa6bce/gradle-jdks-json/src/main/java/com/palantir/gradle/jdks/json/JdksInfoJson.java#L27)

Reply:
> It might be best if this takes the JSON representation that's somewhere else in this repo.

Ah nice, I'll use this then. Didn't know the json representation was in gradle-jdks. I assumed the json representation was in excavator.

> What do we expect to call this? If excavator how will it get all the correct data?

I've done the parsing of the JSON representation in excavator and from there I was calling this method to just write the files.